docs/source/advanced/profiler.rst

```diff
@@ -1,81 +1,108 @@
-.. role:: hidden
-    :class: hidden-section
-
 .. _profiler:
 
-Performance and Bottleneck Profiler
-===================================
+#########
+Profiling
+#########
+
+Profiling your training/testing/inference run can help you understand if there are any bottlenecks in your code. The reports can be generated with ``trainer.fit()`` ``trainer.test()``, ``trainer.validate()`` and ``trainer.predict()`` with the respective actions called.
+
 
-Profiling your training run can help you understand if there are any bottlenecks in your code.
+------------
 
-Built-in checks
----------------
+****************
+Built-in Actions
+****************
 
 PyTorch Lightning supports profiling standard actions in the training loop out of the box, including:
 
 - on_epoch_start
 - on_epoch_end
 - on_batch_start
-- tbptt_split_batch
 - model_forward
 - model_backward
 - on_after_backward
 - optimizer_step
 - on_batch_end
 - training_step_end
 - on_training_end
+- etc...
 
-Enable simple profiling
------------------------
+------------
 
-If you only wish to profile the standard actions, you can set `profiler="simple"`
-when constructing your `Trainer` object.
+*******************
+Supported Profilers
+*******************
 
-.. code-block:: python
+Lightning provides the following profilers:
 
-    trainer = Trainer(..., profiler="simple")
+Simple Profiler
+===============
 
-The profiler's results will be printed at the completion of a training `fit()`.
+If you only wish to profile the standard actions, you can set ``profiler="simple"``. It uses the built-in
+:class:`~pytorch_lightning.profiler.simple.SimpleProfiler`.
 
-.. code-block::
+.. code-block:: python
 
-    Profiler Report
+    # by passing a string
+    trainer = Trainer(..., profiler="simple")
 
-    Action                  |  Mean duration (s)    |  Total time (s)
-    -----------------------------------------------------------------
-    on_epoch_start          |  5.993e-06            |  5.993e-06
-    get_train_batch         |  0.0087412            |  16.398
-    on_batch_start          |  5.0865e-06           |  0.0095372
-    model_forward           |  0.0017818            |  3.3408
-    model_backward          |  0.0018283            |  3.4282
-    on_after_backward       |  4.2862e-06           |  0.0080366
-    optimizer_step          |  0.0011072            |  2.0759
-    on_batch_end            |  4.5202e-06           |  0.0084753
-    on_epoch_end            |  3.919e-06            |  3.919e-06
-    on_train_end            |  5.449e-06            |  5.449e-06
+    # or by passing an instance
+    from pytorch_lightning.profiler import SimpleProfiler
 
+    profiler = SimpleProfiler()
+    trainer = Trainer(..., profiler=profiler)
 
-Advanced Profiling
-------------------
+The profiler's results will be printed at the completion of a training ``trainer.fit()``. Note that there are a lot more actions that will be present in the finalized
```

Comment:
I would put the note below ("the above output is trunceded, more actions will be listed in the ...")